llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.h

```diff
@@ -100,8 +116,9 @@ class LLVM_LIBRARY_VISIBILITY NVPTXDAGToDAGISel : public SelectionDAGISel {
 
   static unsigned GetConvertOpcode(MVT DestTy, MVT SrcTy, LoadSDNode *N);
 
-  NVPTX::Ordering insertMemoryInstructionFence(SDLoc DL, SDValue &Chain,
-                                               MemSDNode *N);
+  std::pair<NVPTX::Ordering, NVPTX::Scope>
```

Comment:
A comment describing what returned `Ordering` and `Scope` have to do with the instruction insertion would be helpful.

Reply:
Done

Reply:
The comment gives an impression that the fence instructions are always inserted, while IIUIC, it inserts fence instructions only *if* such instructions are needed.
I'd rephrase it along the lines of "Inserts an appropriate fence instruction, if it's required to implement particular ordering"

Reply:
Have improved the comment a bit.